beamer/state_machine.py

```diff
@@ -93,6 +93,17 @@ def process_event(event: Event, context: Context) -> HandlerResult:
         raise RuntimeError("Unrecognized event type")
 
 
+def _find_claim_by_fill_hash(context: Context, fill_hash: FillHash) -> Optional[Claim]:
+    for claim in context.claims:
+        request = context.requests.get(claim.request_id)
+        assert request is not None
```

Comment:
Can we encounter claims for non-existent requests?
Or do we discard those immediately?

Reply:
I think this must not happen. As claims and requests come from the same chain they arrive chronologically. The contracts forbid creating a claim for an invalid, expired or withdrawn request (id).
The request object in the agent should never be discarded before expiration or withdrawal. Thus, if a claim event arrives there must be a corresponding request object (otherwise it's a bug)